packages/core/src/request.ts

```diff
@@ -77,9 +77,10 @@ export function transactionToSentryRequest(event: Event, api: API): SentryReques
 
   // the tracestate is stored in bas64-encoded JSON, but envelope header values are expected to be full JS values,
   // so we have to decode and reinflate it
-  let tracestate;
+  let reinflatedTracestate;
   try {
-    tracestate = JSON.parse(base64ToUnicode(encodedTracestate as string));
+    const encodedSentryValue = (tracestate?.sentry as string).replace('sentry=', '');
+    reinflatedTracestate = JSON.parse(base64ToUnicode(encodedSentryValue));
   } catch (err) {
     logger.warn(err);
```

Comment:
This would log `Cannot read property 'replace' of undefined` for situations when there's simply no `tracestate.sentry` which might be confusing to the user.

Reply:
True. Originally I had it as `const encodedSentryValue = tracestate!.sentry!.replace('sentry=', '');`, since if we get to that point and there's no `tracestate.sentry`, things have gone real wrong, but then it was yelling at me about the `!`. But I think maybe I should change it back, to better communicate to the reader that it's not optional.

Or maybe I can write my types in a smarter way... In any case, will fix it one way or another.